File: backend/src/main/java/com/scholarzim/util/ProviderOrgType.java
package com.scholarzim.util;

import java.util.List;
import java.util.Map;


public final class ProviderOrgType {

    public static final String NGO = "NGO";
    public static final String PRIVATE_COMPANY = "PRIVATE_COMPANY";
    public static final String FOUNDATION = "FOUNDATION";
    public static final String GOVERNMENT = "GOVERNMENT";
    public static final String UNIVERSITY = "UNIVERSITY";
    public static final String OTHER = "OTHER";

    public static final List<String> ALL = List.of(
            NGO, PRIVATE_COMPANY, FOUNDATION, GOVERNMENT, UNIVERSITY, OTHER);

    private static final Map<String, String> LABELS = Map.of(
            NGO, "NGO / Non-profit",
            PRIVATE_COMPANY, "Private company",
            FOUNDATION, "Foundation / Trust",
            GOVERNMENT, "Government body",
            UNIVERSITY, "University / College",
            OTHER, "Other registered entity");

    private ProviderOrgType() {
    }

    public static boolean isValid(String value) {
        return value != null && ALL.contains(value);
    }

    public static String label(String code) {
        return LABELS.getOrDefault(code, code);
    }
}

File: app/Support/ProviderOrgType.php
<?php

namespace App\Support;

final class ProviderOrgType
{
    public const NGO = 'NGO';
    public const PRIVATE_COMPANY = 'PRIVATE_COMPANY';
    public const FOUNDATION = 'FOUNDATION';
    public const GOVERNMENT = 'GOVERNMENT';
    public const UNIVERSITY = 'UNIVERSITY';
    public const OTHER = 'OTHER';

    public const ALL = [
        self::NGO,
        self::PRIVATE_COMPANY,
        self::FOUNDATION,
        self::GOVERNMENT,
        self::UNIVERSITY,
        self::OTHER,
    ];

    private const LABELS = [
        self::NGO => 'NGO / Non-profit',
        self::PRIVATE_COMPANY => 'Private company',
        self::FOUNDATION => 'Foundation / Trust',
        self::GOVERNMENT => 'Government body',
        self::UNIVERSITY => 'University / College',
        self::OTHER => 'Other registered entity',
    ];

    private function __construct()
    {
    }

    public static function isValid(?string $value): bool
    {
        return $value !== null && in_array($value, self::ALL, true);
    }

    public static function label(?string $code): string
    {
        return self::LABELS[$code] ?? (string) $code;
    }

    /** code => label, for building selects. */
    public static function options(): array
    {
        return self::LABELS;
    }
}
